src/support/hazard.c

```diff
@@ -103,21 +102,11 @@ __wt_hazard_set(WT_SESSION_IMPL *session, WT_REF *ref, bool *busyp
 
 		/*
 		 * Check if the page state is still valid, where valid means a
-		 * state of WT_REF_MEM and the pointer is unchanged.  (The
-		 * pointer can change, it means the page was evicted between
-		 * the time we set our hazard pointer and the publication.  It
-		 * would theoretically be possible for the page to be evicted
-		 * and a different page read into the same memory, so the
-		 * pointer hasn't changed but the contents have.  That's OK, we
-		 * found this page using the tree's key space, whatever page we
-		 * find here is the page for us to use.)
+		 * state of WT_REF_MEM.
 		 */
 		if (ref->state == WT_REF_MEM) {
-			WT_READ_BARRIER();
-			if (ref->page == page) {
-				++session->nhazard;
-				return (0);
-			}
```

Comment:
I don't think we can omit the read barrier here.

Otherwise, we could publish the ref, check `ref->state == WT_REF_MEM`, then see `ref->page == NULL` because that read is scheduled before the `ref->state` check.